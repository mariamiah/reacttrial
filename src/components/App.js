import React, { Component } from 'react';
import Counter from './counter';
import Nav from  './Nav';
import { Link, Route, BrowserRouter as Router  } from 'react-router-dom';
import '../style.scss';




class App extends Component {
  render() {
    return (
            <div>
                <Nav />
               <div className="card">
                <h5 className="card-header info-color white-text text-center py-4">
                <strong>Sign up</strong>
                </h5>
                <div className="card-body px-lg-5 pt-0">
                <form className="text-center">
                <div className="form-row">
                <div className="col">
                    <div className="md-form">
                        <input type="text" id="materialRegisterFormFirstName" className="form-control" />
                        <label>First name</label>
                    </div>
                </div>
                <div className="col">
                    <div className="md-form">
                        <input type="email" id="materialRegisterFormLastName" className="form-control" />
                        <label>Last name</label>
                    </div>
                </div>
                </div>
            <div className="md-form mt-0">
                <input type="email" id="materialRegisterFormEmail" className="form-control" />
                <label>E-mail</label>
            </div>
            <div className="md-form">
                <input type="password" id="materialRegisterFormPassword" className="form-control" aria-describedby="materialRegisterFormPasswordHelpBlock" />
                <label>Password</label>
                <small id="materialRegisterFormPasswordHelpBlock" className="form-text text-muted mb-4">
                    At least 8 characters and 1 digit
                </small>
            </div>
            <div className="md-form">
                <input type="password" id="materialRegisterFormPhone" className="form-control" aria-describedby="materialRegisterFormPhoneHelpBlock" />
                <label>Phone number</label>
                <small id="materialRegisterFormPhoneHelpBlock" className="form-text text-muted mb-4">
                    Optional - for two step authentication
                </small>
            </div>
            <div className="form-check">
                <input type="checkbox" className="form-check-input" id="materialRegisterFormNewsletter" />
                <label className="form-check-label">Subscribe to our newsletter</label>
            </div>
            <button className="btn btn-outline-info btn-rounded btn-block my-4 waves-effect z-depth-0" type="submit">Sign in</button>
            <p>or sign up with:</p>
            <a type="button" className="btn-floating btn-fb btn-sm">
                <i className="fab fa-facebook-f"></i>
            </a>
            <a type="button" className="btn-floating btn-tw btn-sm">
                <i className="fab fa-twitter"></i>
            </a>
            <a type="button" className="btn-floating btn-li btn-sm">
                <i className="fab fa-linkedin-in"></i>
            </a>
            <a type="button" className="btn-floating btn-git btn-sm">
                <i className="fab fa-github"></i>
            </a>
            <p>By clicking
                <em>Sign up</em> you agree to our
                <a href="" target="_blank">terms of service</a>
              </p> </form>
    </div>

</div>
<Counter />
            </div>
    )
  }
}
export default App;