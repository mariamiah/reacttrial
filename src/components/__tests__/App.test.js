import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router'
import { shallow } from 'enzyme';



test('render a label', ()=>{
    const wrapper = shallow(<label>First name</label>);
    expect(wrapper).toMatchSnapshot();
});

test('should match the snapshot', () => {
  const component = renderer.create(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('should render signup header', ()=>{
  const wrapper = shallow(<App />);
  const header = <strong>Sign up</strong>;
  expect(wrapper.contains(header)).toEqual(true);
})

