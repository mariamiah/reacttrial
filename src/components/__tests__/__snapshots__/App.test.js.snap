// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a label 1`] = `
<label>
  First name
</label>
`;

exports[`should match the snapshot 1`] = `
<div>
  <div>
    <header>
      <li>
        <a
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/articles"
          onClick={[Function]}
        >
          Articles
        </a>
      </li>
    </header>
  </div>
  <div
    className="card"
  >
    <h5
      className="card-header info-color white-text text-center py-4"
    >
      <strong>
        Sign up
      </strong>
    </h5>
    <div
      className="card-body px-lg-5 pt-0"
    >
      <form
        className="text-center"
      >
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <div>
              <input
                className="form-control"
                id="materialRegisterFormLastName"
                type="email"
              />
              <label>
                First name
              </label>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="md-form"
            >
              <input
                className="form-control"
                id="materialRegisterFormLastName"
                type="email"
              />
              <label>
                Last name
              </label>
            </div>
          </div>
        </div>
        <div
          className="md-form mt-0"
        >
          <input
            className="form-control"
            id="materialRegisterFormEmail"
            type="email"
          />
          <label>
            E-mail
          </label>
        </div>
        <div
          className="md-form"
        >
          <input
            aria-describedby="materialRegisterFormPasswordHelpBlock"
            className="form-control"
            id="materialRegisterFormPassword"
            type="password"
          />
          <label>
            Password
          </label>
          <small
            className="form-text text-muted mb-4"
            id="materialRegisterFormPasswordHelpBlock"
          >
            At least 8 characters and 1 digit
          </small>
        </div>
        <div
          className="md-form"
        >
          <input
            aria-describedby="materialRegisterFormPhoneHelpBlock"
            className="form-control"
            id="materialRegisterFormPhone"
            type="password"
          />
          <label>
            Phone number
          </label>
          <small
            className="form-text text-muted mb-4"
            id="materialRegisterFormPhoneHelpBlock"
          >
            Optional - for two step authentication
          </small>
        </div>
        <div
          className="form-check"
        >
          <input
            className="form-check-input"
            id="materialRegisterFormNewsletter"
            type="checkbox"
          />
          <label
            className="form-check-label"
          >
            Subscribe to our newsletter
          </label>
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          SignUp
        </button>
        <p>
          or sign up with:
        </p>
        <a
          className="btn-floating btn-fb btn-sm"
          type="button"
        >
          <i
            className="fab fa-facebook-f"
          />
        </a>
        <a
          className="btn-floating btn-tw btn-sm"
          type="button"
        >
          <i
            className="fab fa-twitter"
          />
        </a>
        <a
          className="btn-floating btn-li btn-sm"
          type="button"
        >
          <i
            className="fab fa-linkedin-in"
          />
        </a>
        <a
          className="btn-floating btn-git btn-sm"
          type="button"
        >
          <i
            className="fab fa-github"
          />
        </a>
        <p>
          By clicking
          <em>
            Sign up
          </em>
           you agree to our
          <a
            href=""
            target="_blank"
          >
            terms of service
          </a>
        </p>
         
      </form>
    </div>
  </div>
  <div>
    <h1>
      My Counter
    </h1>
    0
    <button
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      onClick={[Function]}
    >
      Decrement
    </button>
  </div>
</div>
`;
